web/server/codechecker_server/api/report_server.py

```diff
@@ -2861,6 +2863,187 @@ def getCheckerCounts(self, run_ids, report_filter, cmp_data, limit,
+                    subquery.c.severity,
+                    subquery.c.run_id,
+                    case(
+                        [
+                            (
+                                subquery.c.detection_status.in_(list(map(
+                                    detection_status_str,
+                                    (DetectionStatus.OFF,
+                                     DetectionStatus.UNAVAILABLE)
+                                ))),
+                                False
+                            )
+                        ],
+                        else_=True
+                    ),
+                    case(
+                        [
+                            (
+                                and_(
+                                    subquery.c.detection_status.in_(list(map(
+                                        detection_status_str,
+                                        (DetectionStatus.NEW,
+                                         DetectionStatus.UNRESOLVED,
+                                         DetectionStatus.REOPENED)
+                                    ))),
+                                    subquery.c.review_status.in_(list(map(
+                                        review_status_str,
+                                        (API_ReviewStatus.UNREVIEWED,
+                                         API_ReviewStatus.CONFIRMED))))
+                                ),
+                                True
+                            )
+                        ],
+                        else_=False
+                    )
+                )
+            )
+
+            checker_stats = {}
+            all_run_id = [runId for runId, _ in max_run_histories.all()]
+            for checker_id, \
+                checker_name, \
+                analyzer_name, \
+                severity, \
+                run_ids, \
+                is_enabled, \
+                is_opened, \
+                cnt \
+                    in query.all():
+                checker_stat = checker_stats[checker_id] \
+                    if checker_id in checker_stats \
+                    else CheckerInfo(
+                        checkerName=checker_name,
+                        analyzerName=analyzer_name,
+                        enabled=[],
+                        disabled=all_run_id,
+                        severity=severity,
+                        closed=0,
+                        outstanding=0
+                    )
+
+                if is_enabled:
+                    for r in (run_ids.split(",")
+                              if type(run_ids) is str
+                              else [run_ids]):
+                        run_id = int(r)
+                        if run_id not in checker_stat.enabled:
+                            checker_stat.enabled.append(run_id)
+                        if run_id in checker_stat.disabled:
+                            checker_stat.disabled.remove(run_id)
```

Comment:
Perhaps it would be useful to have `set`s here for quicker lookup, and only convert them to a list at the end before the value of `checker_stat` is completely calculated. (Also, wouldn't that help with returning a sorted list to the API request as well?)

Reply:
Storno, because the data structure is updated often but it's directly in the Thrift type so it would need back-and-forth conversions.